File: UIExporter/Test/TestUi.cs
using System;
using UnityEditor;
using UnityEngine;

/**
* Editor Script Component
* @class TestUi
*/
[Babylon(Class="PROJECT.TestUi"), AddComponentMenu("Scripts/My Project/TestUi")]
public class TestUi : EditorScriptComponent
{
    /* Add Editor Properties To Script Component */
    // Example: [Tooltip("Example hello world property")]
    // Example: [Auto] public string helloWorld = "Hello World";

	/* [Serializable, HideInInspector] public string exportProperty = null; */
    public override void OnUpdateProperties(Transform transform, SceneExporterTool exporter)
    {
        // Example: this.helloWorld = "Update Hello World";
    }
}

// Optional Script Component Custom Editor Class
[CustomEditor(typeof(TestUi)), CanEditMultipleObjects]
public class TestUiEditor : Editor
{
    public void OnEnable()
    {
        TestUi owner = (TestUi)target;
    }
    public override void OnInspectorGUI()
    {
        base.OnInspectorGUI();
    }
}

File: UIExporter/Test/TestUi.ts
module PROJECT {
  /**
   * Babylon Script Component
   * @class TestUi
   */
  export class TestUi extends BABYLON.ScriptComponent {
    // Example: private helloWorld:string = "Hello World";

    protected awake(): void {
      /* Init component function */
    }

    protected start(): void {
      new AdvancedTextureUi().create();
    }

    protected ready(): void {
      /* Execute when ready function */
    }

    protected update(): void {
      /* Update render loop function */
    }

    protected late(): void {
      /* Late update render loop function */
    }

    protected after(): void {
      /* After update render loop function */
    }

    protected fixed(): void {
      /* Fixed update physics step function */
    }

    protected destroy(): void {
      /* Destroy component function */
    }
  }
}
